lib/access/accessSchema.rb

```diff
@@ -62,86 +62,607 @@ def is_withdrawn(obj)
+  description "Ordering for item list results"
+  value("ADDED_ASC", "Date added to eScholarship, oldest to newest")
+  value("ADDED_DESC", "Date added to eScholarship, newest to oldest")
+  value("PUBLISHED_ASC", "Date published, oldest to newest")
+  value("PUBLISHED_DESC", "Date published, newest to oldest")
+  value("UPDATED_ASC", "Date updated in eScholarship, oldest to newest")
+  value("UPDATED_DESC", "Date updated in eScholarship, newest to oldest")
+end
+
+###################################################################################################
+class ItemStatusEnum < GraphQL::Schema::Enum
+  graphql_name "ItemStatus"
+  description "Publication status of an Item (usually PUBLISHED)"
+  value("EMBARGOED", "Currently under embargo (omitted from queries)")
+  value("EMPTY", "Item was published but has no link or files (omitted from queries)")
+  value("PUBLISHED", "Normal published item")
+  value("WITHDRAWN", "Item was withdrawn (omitted from queries)")
+  value("PENDING", "Item is still pending")
+end
+
+###################################################################################################
+class ItemTypeEnum < GraphQL::Schema::Enum
+  graphql_name "ItemType"
+  description "Publication type of an Item (often ARTICLE)"
+  value("ARTICLE", "Normal article, e.g. a journal article")
+  value("CHAPTER", "Chapter within a book/monograph")
+  value("ETD", "Electronic thesis/dissertation")
+  value("MONOGRAPH", "A book / monograph")
+  value("MULTIMEDIA", "Multimedia (e.g. video, audio, etc.)")
+  value("NON_TEXTUAL", "Other non-textual work")
+end
+
+###################################################################################################
+class AuthorIDType < GraphQL::Schema::Object
+  graphql_name "AuthorID"
+  description "Author identifier, e.g. escholarship, ORCID, other."
+
+  field :id, String, "The identifier string" do
+    def resolve(obj, args, ctx) 
+      obj = obj.object
+      obj['id'] 
+    end
+  end
+
+  field :scheme, AuthorIDSchemeEnum, "The scheme under which the identifier was minted", null: false do
+    def resolve(obj, args, ctx)
+      obj = obj.object
+      case obj['type']
+        when 'ARK';   "ARK"
+        when 'ORCID'; "ORCID"
+        else          "OTHER_ID"
+      end
+    end
+  end
+
+  field :subScheme, String, "If scheme is OTHER_ID, this will be more specific" do
+    def resolve(obj, args, ctx)
+     obj = obj.object
+      case obj['type']
+        when 'ARK'; nil
+        when 'ORCID'; nil
+        else obj['type']
+      end
+    end
+  end
+end
+
+###################################################################################################
+class NamePartsType < GraphQL::Schema::Object
+  graphql_name "NameParts"
+  description "Individual access to parts of the name, generally only used in special cases"
+  field :name, String, "Combined name parts; usually 'lname, fname'", null: false do
+    def resolve(obj, args, ctx) 
+      obj = obj.object
```

Comment:
Looking at the [graphql-ruby docs on field resolution](https://graphql-ruby.org/fields/introduction#field-resolution) it looks like you should be able to do something like
```
field :name, String, "Combined name parts; usually 'lname, fname'", null: false, hash_key: "name"
```

Feels like a lot of duplication could be eliminated with that approach. Are our objects structured differently than the docs seem to assume they will be? If so, do we know why?

Reply:
@martinhaye knows best on why for the current structure. I have tried to minimize changes. Graphql2 documents recommend newer resolve way with separate classes. If the team wants to align with the latest Graphql recommendations - that could be a project after AL2023 work.

Reply:
Been too long for me to remember why I did things the way I did. Modernizing sounds like a good idea, and also work.

Reply:
Yeah, considering the plan to push more and more functionality into the GraphQL API as a way of further isolating the parts of eScholarship. I definitely think we want to invest in

1. keeping the implementation modern and up to date
2. figuring out how to write tests for it so we can merge changes to keep it up to date with less worry that we're going to break something.

Agree with @myucekul that is should be separated fro this upgrade.